<launch>
    <arg name="use_sim_time" default="false"/>
    <arg name="timeout" default="350"/>
    <arg name="goal_weight" default="350"/>
    <arg name="cost_weight" default="350"/>
    <arg name="line_of_sight_distance" default="1"/>
    <arg name="map" default="$(find theta_star_2d)/octomap_full.bt"/>
    <arg name="batch_mode" default="false"/>
    <param name="use_sim_time" value="$(arg use_sim_time)"/>

    <node pkg="theta_star_2d" type="global_planner_node" name="global_planner_node" output="screen" required="true">
        <param name="map_path"               value="$(arg map)"/>
        <param name="mode3d"                 value="true"/>
        <param name="map_resolution"         value="0.1"/>
        <param name="map_h_inflaction"       value="0.3"/>
        <param name="map_v_inflaction"       value="0.3"/>
        <param name="z_weight_cost"          value="1"/>
        <param name="goal_weight"            value="$(arg goal_weight)"/>
        <param name="cost_weight"            value="$(arg cost_weight)"/>
        <param name="line_of_sight_distance" value="$(arg line_of_sight_distance)"/>
        <param name="publish_point_cloud"    value="true"/>
        <remap from="points" to="/grid3d/map_point_cloud"/>
        <param name="ws_x_max"   value="110"/>
        <param name="ws_x_min"   value="75"/>
        <param name="ws_y_max"   value="140"/>
        <param name="ws_y_min"   value="100"/>
        <param name="ws_z_max"   value="10"/>
        <param name="ws_z_min"   value="0"/>
        <param name="show_config" value="true"/>
        <param name="debug" value="true"/>
        <param name="world_frame" value="map"/>
        <param name="robot_base_frame" value="base_link"/>
        <param name="wait_for_servers" value="false"/>
        <param name="timeout"   value="$(arg timeout)"/>
    </node>

    <node pkg="static_tf_dyncfg" name="static_tf_dyncfg_node" type="static_tf_dyncfg_node" args="map base_link 102 106 3 0 0 0"/>
    <node pkg="rqt_publisher" name="rqt_publisher" type="rqt_publisher" unless="$(arg batch_mode)"/>
    <node pkg="rqt_reconfigure" name="rqt_reconfigure" type="rqt_reconfigure" unless="$(arg batch_mode)"/>

    <node pkg="rviz" name="rviz" type="rviz" args="-d $(find theta_star_2d)/viz/3dcost_test.rviz"/>

</launch>
